pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission_test.go

```diff
@@ -0,0 +1,279 @@
+			Expect(err).NotTo(HaveOccurred())
+
+			handler = extensionresources.New(logger)
+			Expect(admission.InjectDecoderInto(decoder, handler)).To(BeTrue())
+
+			request = nil
+		})
+
+		AfterEach(func() {
+			ctrl.Finish()
+		})
+
+		Context("ignored requests", func() {
+			It("should ignore DELETE operations", func() {
+				request = newRequest(admissionv1.Delete, "dnsrecords", "", "")
+				expectAllowed(handler.Handle(ctx, *request), ContainSubstring("operation is not CREATE or UPDATE"))
+			})
+
+			It("should ignore CONNECT operations", func() {
+				request = newRequest(admissionv1.Connect, "dnsrecords", "", "")
+				expectAllowed(handler.Handle(ctx, *request), ContainSubstring("operation is not CREATE or UPDATE"))
+			})
+
+			It("should ignore different resources", func() {
+				request = newRequest(admissionv1.Create, "customresourcedefinitions", "", "")
+				expectAllowed(handler.Handle(ctx, *request), ContainSubstring("validation not found for the given resource"))
+			})
+		})
+
+		Context("create resources", func() {
+			var resources = []dummyResource{
+				{
+					kind: "backupbuckets",
+					obj:  fmt.Sprintf(backupBucketFmt, "", "gcp", "backupprovider"),
+				},
+				{
+					kind: "backupentries",
+					obj:  fmt.Sprintf(backupEntryFmt, "", "gcp", "backupentry"),
+				},
+				{
+					kind: "bastions",
+					obj:  fmt.Sprintf(bastionFmt, "", "gcp", "shoot-ref-2"),
+				},
+				{
+					kind: "controlplanes",
+					obj:  fmt.Sprintf(controlPlaneFmt, "", "gcp", "cloudprovider"),
+				},
+				{
+					kind: "dnsrecords",
+					obj:  fmt.Sprintf(dnsrecordFmt, "", "dnsrecord-external", "A"),
+				},
+				{
+					kind: "extensions",
+					obj:  fmt.Sprintf(extensionsFmt, "", "gcp", "seed-gcp"),
+				},
+				{
+					kind: "infrastructures",
+					obj:  fmt.Sprintf(infrastructureFmt, "", "gcp", "seed-gcp"),
+				},
+				{
+					kind: "networks",
+					obj:  fmt.Sprintf(networksFmt, "", "calico", "seed-gcp"),
+				},
+				{
+					kind: "operatingsystemconfigs",
+					obj:  fmt.Sprintf(operatingsysconfigFmt, "", "gcp", "seed-gcp"),
+				},
+				{
+					kind: "workers",
+					obj:  fmt.Sprintf(workerFmt, "", "gcp", "seed-gcp"),
+				},
+			}
+
+			It("should create successfully the resources", func() {
+				for _, r := range resources {
+					request = newRequest(admissionv1.Create, r.kind, r.obj, "")
+					expectAllowed(handler.Handle(ctx, *request), ContainSubstring("validation successful"), resourceToId(request.Resource))
+				}
+			})
+
+			It("decoding of new dns record should fail", func() {
+				dns := fmt.Sprintf(dnsrecordFmt, `"resourceVersion" "1",`, "name", "A")
+				request = newRequest(admissionv1.Create, "dnsrecords", dns, "")
+				expectErrored(handler.Handle(ctx, *request), ContainSubstring("could not decode ar"), BeEquivalentTo(http.StatusUnprocessableEntity))
+			})
+		})
+
+		Context("update resources", func() {
+			BeforeEach(func() {
+
+				resources = []dummyResource{
```

Comment:
Instead of this dummyResource slice, isn't better to use a table test? As far as I see the only point behind it is that it simulates a table test.